cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaExporterReconciler.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly;
+
+import io.fabric8.kubernetes.api.model.LocalObjectReference;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.strimzi.api.kafka.model.Kafka;
+import io.strimzi.operator.cluster.model.Ca;
+import io.strimzi.operator.cluster.model.ClusterCa;
+import io.strimzi.operator.cluster.model.ImagePullPolicy;
+import io.strimzi.operator.cluster.model.KafkaExporter;
+import io.strimzi.operator.cluster.model.KafkaVersion;
+import io.strimzi.operator.cluster.model.ModelUtils;
+import io.strimzi.operator.common.Annotations;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.operator.resource.DeploymentOperator;
+import io.strimzi.operator.common.operator.resource.ReconcileResult;
+import io.strimzi.operator.common.operator.resource.SecretOperator;
+import io.strimzi.operator.common.operator.resource.ServiceAccountOperator;
+import io.vertx.core.Future;
+
+import java.util.Date;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * Class used for reconciliation of Kafka Exporter. This class contains both the steps of the Kafka Exporter
+ * reconciliation pipeline and is also used to store the state between them.
+ */
+public class KafkaExporterReconciler {
```

Comment:
Compared to the naming in operator sdks (the Go [1] and the Java [2] ones) maybe we should call this as `KafkaExporterController`? It takes care of everything related to the Kafka Exporter inside the overall Strimzi cluster operator. The only difference with "controllers" in other operators is that this one doesn't really handle a `KafkaExporter` custom resource but more a part of the `Kafka` one.

[1] https://github.com/operator-framework/operator-sdk-samples/blob/master/go/memcached-operator/controllers/memcached_controller.go
[2] https://github.com/java-operator-sdk/samples/blob/main/webserver/src/main/java/io/javaoperatorsdk/operator/sample/WebServerController.java

Reply:
It does not really follow the Kubernetes controller pattern too much. It follows our _assembly operator_ pattern. So I intentionally wanted to avoid calling it a controller. I was thinking about calling it `KafkaExporterAssemblyOperator` ... but it might be confusing with not being a high level `*AssemblyOperator` as the other ones. So I chose `Reconciler`. I'm open to discussion about the name -> but I do not want to call it Controller.